From 584193cc2defc31b5a13f7ef1bd6d2376d6606e7 Mon Sep 17 00:00:00 2001
From: Mike Hommey <glandium@debian.org>
Date: Sat, 21 Jun 2008 03:09:21 +0200
Subject: [PATCH 19/40] Add another preferences directory for applications: preferences/syspref

It was existing in previous versions of iceweasel as a symlink to
/etc/iceweasel/pref.

This has the side effect to make these preferences there work again, and
to disable the "set as default browser" dialog.
---
 toolkit/xre/nsXREDirProvider.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/toolkit/xre/nsXREDirProvider.cpp b/toolkit/xre/nsXREDirProvider.cpp
index a60d45e..c80d6f1 100644
--- a/toolkit/xre/nsXREDirProvider.cpp
+++ b/toolkit/xre/nsXREDirProvider.cpp
@@ -621,6 +621,7 @@ nsXREDirProvider::LoadAppBundleDirs()
 }
 
 static const char *const kAppendPrefDir[] = { "defaults", "preferences", nsnull };
+static const char *const kAppendSysPrefDir[] = { "defaults", "syspref", nsnull };
 
 #ifdef DEBUG_bsmedberg
 static void
@@ -677,6 +678,7 @@ nsXREDirProvider::GetFilesInternal(const char* aProperty,
     LoadAppDirIntoArray(mXULAppDir, kAppendPrefDir, directories);
     LoadDirsIntoArray(mAppBundleDirectories,
                       kAppendPrefDir, directories);
+    LoadAppDirIntoArray(mXULAppDir, kAppendSysPrefDir, directories);
 
     rv = NS_NewArrayEnumerator(aResult, directories);
   }
-- 
1.7.0.rc0.69.g2ee80

